### ISSUE TITLE
[Proposal] Allow to exclude dependency namespaces in Credo.Check.Refactor.ModuleDependencies

### ISSUE THREAD

**Issue by Eiji7** [rrrene/credo-proposals#106]
`credo` prints messages about too many module dependencies that are expected. Unfortunately I'm unable to exclude them (as I can only exclude file path or whole module having many dependencies). I would like to check if except said group my module does not have too much dependencies.

The argument for it is to work with a group of sub-modules (modules in same namespace) instead of single and huge one.  In my case there are 8 sub-modules, so I don't need a lot of effort to cause `credo`'s message. I'm working on custom DSL (think about `Ecto.Schema`) and either I have a huge module with many dependencies or I split it based on features (like each DSL macro).

I think about support for 3rd argument in below code:
https://github.com/rrrene/credo/blob/91502cec785f8a12633851384d6ec83c0c549d39/lib/credo/check/refactor/module_dependencies.ex#L87

and add `reject_namespaces/2` like:
```elixir
  defp reject_namespaces(dependencies, namespaces) do
    Enum.reject(dependencies, &keep?(&1, namespaces))
  end
```

Current workaround is to put below line at the start of file defining module with many dependencies:
```elixir
# credo:disable-for-this-file Credo.Check.Refactor.ModuleDependencies
```

This feature should be simple to implement and I think it would be very helpful. What do you think about it?

**Comment by rrrene** [comment]
@Eiji7 Hi, I transfered this issue to the proposal repo, as described in Credo's issue template 👍

**Comment by rrrene** [comment]
> I would like to check if except said group my module does not have too much dependencies.

Okay, for my understanding: You have a module - let's call it `Foo.API` that reaches out in many other modules, but this is by design. Like `Ecto.Schema` it is sort of a fascade for the other module's functionalities. And now you would like to be able to tell the check: "Tell me if `Foo.API` is reaching out into too many other modules, but ignore a given list of modules, because it is supposed to reach out into these modules."

Correct?

**Comment by Eiji7** [comment]
Yes, to be specific it would be any module starting with `Foo.API.`

**Comment by Eiji7** [comment]
Here is a code example from `dev_joy` package:

> ```elixir
>   # …
>   alias DevJoy.Scene.Asset
>   alias DevJoy.Scene.Challenge
>   alias DevJoy.Scene.Choice
>   alias DevJoy.Scene.Condition
>   alias DevJoy.Scene.Dialog
>   alias DevJoy.Scene.Menu
>   alias DevJoy.Scene.Note
>   # …
> 
>   @type t :: Asset.t() | Challenge.t() | Condition.t() | Dialog.t() | Menu.t() | Note.t() | struct()
>   # …
>   @type t_acc :: t | Choice.t()
> ```
> Source: https://gitlab.com/ramurix-software/dev-joy/-/blob/main/lib/dev_joy/api/item.ex?ref_type=heads#L125

There are few other modules like [`DevJoy.Scene`](https://gitlab.com/ramurix-software/dev-joy/-/blob/main/lib/dev_joy/scene/scene.ex?ref_type=heads) which defines DSL macros for all scene elements. Obviously **tests** for those files also need said dependencies.
